### PR TITLE
Include text-based inputs inside X-data when translating #8009

### LINKED ISSUE #8009: Include text-based inputs inside X-data when translating
- Pass XData and it's schema via event to AI apps
- Update XData text inputs on translation

### PR DESCRIPTION
**Pull request by ashklianko**
Requires https://github.com/enonic/lib-admin-ui/pull/3770

### CODE REVIEW

File: modules/lib/src/main/resources/assets/js/app/ai/AI.ts [hunk 1]
```diff
@@ -198,6 +210,13 @@ export class AI {
         });
     }
 
+    private createXData(): ContentXData[] {
+        return this.content ? this.content.getAllExtraData().map((extraData: ExtraData) => ({
+            name: extraData.getName().toString(),
+            fields: extraData.getData().toJson(),
+        })) : [];
```
Comment (edloidas) [inline modules/lib/src/main/resources/assets/js/app/ai/AI.ts]:
```suggestion
        return this.content?.getAllExtraData().map((extraData: ExtraData) => ({
            name: extraData.getName().toString(),
            fields: extraData.getData().toJson(),
        })) ?? [];
```

File: modules/lib/src/main/resources/assets/js/app/ai/AI.ts [hunk 2]
```diff
@@ -217,6 +236,15 @@ export class AI {
         };
     }
 
+    private createXDataSchemas(): ContentSchema[] {
+        return this.xDataSchemas ? this.xDataSchemas.map(xData => {
+            return {
+                form: xData.toForm().toJson(),
+                name: xData.getName(),
+            };
+        }): [];
```
Comment (edloidas) [inline modules/lib/src/main/resources/assets/js/app/ai/AI.ts]:
```suggestion
        return this.xDataSchemas?.map(xData => {
            return {
                form: xData.toForm().toJson(),
                name: xData.getName(),
            };
        }) ?? [];
```

File: modules/lib/src/main/resources/assets/js/app/ai/AI.ts [hunk 3]
```diff
@@ -253,4 +281,14 @@ export class AI {
...
     }
+
+    private getAiHelperByXData(xDataName: string, path: string): AiHelper | undefined {
+        const xDataStepForm = XDataWizardStepForm.getXDataWizardStepForm(xDataName);
+
+        if (xDataStepForm) {
+            return AiHelper.getAiHelpersByParent(xDataStepForm).find(aiHelper => aiHelper.getDataPath() === path);
+        }
+
+        return undefined;
```
Comment (edloidas) [inline modules/lib/src/main/resources/assets/js/app/ai/AI.ts]:
```suggestion
        const xDataStepForm = XDataWizardStepForm.getXDataWizardStepForm(xDataName);
        return xDataStepForm ? AiHelper.getAiHelpersByParent(xDataStepForm).find(aiHelper => aiHelper.getDataPath() === path) : undefined;
```

**Review by edloidas — CHANGES_REQUESTED** [review]
Nice! Just a few stylistic improvements.

**Review by alansemenov — APPROVED** [review]

**Review by edloidas — APPROVED** [review]

**Pull request merged**